services/venice-router/src/main/java/com/linkedin/venice/router/stats/RouterHttpRequestStats.java

```diff
@@ -55,6 +55,9 @@ public class RouterHttpRequestStats extends AbstractVeniceHttpStats {
   private final Sensor readQuotaUsageSensor;
   private final Sensor inFlightRequestSensor;
   private Sensor keySizeSensor;
+
```

Comment:
Also `keySizeSensor` will be `null` today if `isKeyValueProfilingEnabled` is `disabled`..
I wonder why today, there is no NPE thrown when calling function: `recordKeySizeInByte`, which doesn't do a null check..

Reply:
there is a null check before calling the key size record.

Reply:
Can you share the code link?

Reply:
https://github.com/linkedin/venice/blob/8e0f0ecd0f4c69dbedcd74edc3adb7b1ebef486c/services/venice-router/src/main/java/com/linkedin/venice/router/api/VenicePathParser.java#L146

Reply:
I see.
The check in `VenicePathParser` is not great, and we should do a `null` check in `RouterHttpRequestStats` in function of `recordKeySizeInByte`.